Dockerfile

```diff
@@ -8,14 +8,22 @@ WORKDIR /srv
 
 COPY .yar[n] ./.yarn
 COPY package.json yarn.lock .yarnrc.yml ./
-RUN yarn config set httpProxy $HTTP_PROXY
-RUN yarn config set httpsProxy $HTTPS_PROXY
+RUN if [ -n "$HTTP_PROXY" ]; then \
+      yarn config set httpProxy $HTTP_PROXY; \
+    fi
+RUN if [ -n "$HTTPS_PROXY" ]; then \
+      yarn config set httpsProxy $HTTPS_PROXY; \
+    fi
+
 RUN yarn install
 
 # Build stage: Run "yarn run build-js"
 # ===
 FROM yarn-dependencies AS build-js
 ADD . .
+RUN if [ -n "$HTTP_PROXY" ] || [ -n "$HTTPS_PROXY" ]; then \
+      cp public/config.demo.js public/config.js; \
+    fi
```

Comment:
I wasn't able to access the Jenkins environment variables from here, so, checking if either of `HTTP_PROXY` or `HTTPS_PROXY` is defined in order to see if it is running the demo, is the best I could do.

Also, I couldn't copy to `public/config.local.js`, as the file is not being parsed. I think it might be due to the fact that `process.env.NODE_ENV` is set to `production` when the image is built, thus parsing `public/config.local.js` in `index.html` is skipped.